package com.Chapter10ExceptionHandling.programs;

//An example of nested try statements.

public class NestTry {
	public static void main(String[] args) {
		try
		{
			int a = 3;
			
			/* If no command-line are present,
			 * the following statement will generate
			 * a divide-by-zero exception. */
			
			int b = 42 / a;
			
			System.out.println("a = " + a);
			System.out.println("b = " + b);
			
			try
			{// Nested try block
				/* If one command-line arg is used,
				 * then a divide-by-zero exception
				 * will be generated by the following code. */
				
				if(a == 1) a = a/(a-a); //division by zero
				
				/*If two command-line args are used,
				 * then generate an out of bunds exception.*/
				
				if(a == 3)
				{
					int c[] = { 1 };
					c[42]	= 99; // Generate an out of bounds exception
				}
			}
				
				catch(ArrayIndexOutOfBoundsException e)
				{
					System.out.println("Array index out of bounds: " + e);
				}
				
			}
		catch(ArithmeticException e)
		{
			System.out.println("Divide by zero: " + e);
		}
		}
	}

